api/client/build.go

```diff
@@ -308,13 +322,21 @@ func (cli *DockerCli) CmdBuild(args ...string) error {
 		if jerr.Code == 0 {
 			jerr.Code = 1
 		}
+		if *suppressOutput {
+			fmt.Fprintln(cli.err, outBuf, sopts.out)
+		}
 		return Cli.StatusError{Status: jerr.Message, StatusCode: jerr.Code}
 	}
-
 	if err != nil {
 		return err
 	}
 
+	// Everything worked so if -q was provided the output from the daemon
+	// should be just the image ID and we'll print that to stdout.
+	if *suppressOutput {
+		fmt.Fprintln(cli.out, sopts.out)
```

Comment:
yes, it will output the *full* image ID. You can see [here](https://github.com/ripcurld00d/docker/blob/cli_ninja_build/integration-cli/docker_cli_build_test.go#L4883) that I check that, when the build is successful, the output is a 64 long  alphanumeric characters string.